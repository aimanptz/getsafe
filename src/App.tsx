import React from 'react';
import './App.css';

const AppRouter = React.lazy(() => import('./router'));

const App = () => {
	return (
		<React.Suspense fallback={<p>Loading...</p>}>
			<AppRouter />
		</React.Suspense>
	);
};

export default App;
