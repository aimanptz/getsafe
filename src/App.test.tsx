import React from 'react';
import { render, waitForElementToBeRemoved, waitFor } from '@testing-library/react';
import App from './App';

describe('test App.tsx', () => {
	test('test App component snapshot', async () => {
		const { asFragment, getByText } = render(<App />);

		await waitForElementToBeRemoved(getByText('Loading...'));

		expect(asFragment()).toMatchSnapshot();
	});

	test('renders link to purchase flow', async () => {
		const { getByText } = render(<App />);

		await waitFor(() => {
			expect(getByText('Get started!')).toBeInTheDocument();
		});
	});
});
