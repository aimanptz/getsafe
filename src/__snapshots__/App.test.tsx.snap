// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test App.tsx test App component snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
    style=""
  >
    <header
      class="App-header"
    >
      <img
        alt="logo"
        class="App-logo"
        src="logo.svg"
      />
    </header>
    <p>
      Welcome to Getsafe's Developer Insurance
    </p>
    <a
      href="/purchase/insurance"
    >
      Get started!
    </a>
  </div>
</DocumentFragment>
`;
